controllers/configmap_controller.go

```diff
@@ -170,6 +176,43 @@ func (r *ConfigMapReconciler) reconcileServices(ctx context.Context, windowsServ
 	return nil
 }
 
+// removeOutdatedServicesConfigMaps deletes any outdated services ConfigMaps, if all nodes have moved past that version
+func (r *ConfigMapReconciler) removeOutdatedServicesConfigMaps(ctx context.Context) error {
+	watchNamespaceCMs := &core.ConfigMapList{}
+	if err := r.client.List(ctx, watchNamespaceCMs, &client.ListOptions{Namespace: r.watchNamespace}); err != nil {
+		return err
+	}
+	configMapsToDelete := []core.ConfigMap{}
+	for _, cm := range watchNamespaceCMs.Items {
+		// A possibly outdated services ConfigMap has the name "windows-services-*" where * != current WMCO version
+		if strings.HasPrefix(cm.Name, servicescm.NamePrefix) && cm.Name != servicescm.Name {
+			configMapsToDelete = append(configMapsToDelete, cm)
+		}
+	}
+	if len(configMapsToDelete) == 0 {
+		// Nothing to clean up
+		return nil
+	}
+
+	nodes := &core.NodeList{}
+	if err := r.client.List(ctx, nodes, client.MatchingLabels{core.LabelOSStable: "windows"}); err != nil {
+		return err
+	}
+	// Check if there are any nodes that require upgrade to make sure configMapsToDelete are actually outdated
```

Comment:
I feel this logic is out of the scope of `removeOutdatedServicesConfigMaps()`, first we should check if all the preconditions are met, then invoke the cleanup